src/db.py

```diff
@@ -240,22 +236,15 @@ def create_search_task(
         return job
 
     def get_job_matches(
-        self, job: JobId, offset: int = 0, limit: Optional[int] = None
+        self, job_id: JobId, offset: int = 0, limit: Optional[int] = None
     ) -> MatchesSchema:
-        if limit is None:
-            end = -1
-        else:
-            end = offset + limit - 1
-        meta = self.redis.lrange(f"meta:{job}", offset, end)
-        matches = [json.loads(m) for m in meta]
-        for match in matches:
-            # Compatibility fix for old jobs, without sha256 metadata key.
-            if "sha256" not in match["meta"]:
-                match["meta"]["sha256"] = {
-                    "display_text": "0" * 64,
-                    "hidden": True,
-                }
-        return MatchesSchema(job=self.get_job(job), matches=matches)
+        with Session(self.engine) as session:
+            job = self.__get_job(session, job_id)
+            if limit is None:
+                matches = job.matches[offset:]
```

Comment:
This seems like an easy way out but we should probably do the pagination on the db side, right?
In cases where we'll be getting thousands of results it actually makes sense to make it gotta go fast.

Reply:
Oh, it's a single array field. :hmmm:

Reply:
Probably not worth it then

Reply:
Nah, actually currently job.matches is a backreference to a list of Match objects (and every match object has an unrelated `matches` array... Duh, didn't realise until now. I already planned to rename i to `matched_rules`, but not all at once). Yeah, it makes a lot of sense to do this on the DB side, I was just lazy.

I've created an issue for it: #372.